src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -48,18 +96,18 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// more messages, and a new queue will need to be created.
     /// </para>
     /// </remarks>
-    internal partial class RequestExecutionQueue
+    public abstract partial class RequestExecutionQueue<RequestContextType> : IRequestExecutionQueue<RequestContextType> where RequestContextType : struct
     {
-        private readonly WellKnownLspServerKinds _serverKind;
-        private readonly ImmutableArray<string> _supportedLanguages;
-        private readonly ILspLogger _logger;
-        private readonly LspServices _lspServices;
+        internal readonly string _serverKind;
```

Comment:
I'm making these more visible for now because I'm playing with the idea of including a generallized version of RequestExecutionQueue with CLaSP. If that doesn't pan out these ought to go back to private.

Reply:
I mentioned something similar in my comments. I think the request routing, invocation, management etc. make a lot of sense in CLaSP while simultaneously respecting one of your goals of "everything can be replaced"

Reply:
100% - the library isn't super interesting without these bits